contracts/StablecoinConverter.sol

```diff
@@ -173,12 +182,62 @@ contract StablecoinConverter is EpochTokenLocker {
         orders[owner][orderId].sellAmount = newSellAmount;
     }
 
-    function writeCurrentPrices(
+    function updateCurrentPrices(
         uint128[] memory prices,  //list of prices for touched token only
         uint16[] memory tokenIdsForPrice  // price[i] is the price for the token with tokenID tokenIdsForPrice[i]
     ) internal {
+        for (uint i = 0; i < previousSolutionTokenIdForPrice.length; i++) {
+            currentPrices[previousSolutionTokenIdForPrice[i]] = 0;
+        }
         for (uint i = 0; i < tokenIdsForPrice.length; i++) {
             currentPrices[tokenIdsForPrice[i]] = prices[i];
         }
     }
+
+    function documentTrades(
+        uint batchIndex,
+        address[] memory owners,  //tradeData is submitted as arrays
+        uint16[] memory orderIds,
+        uint128[] memory volumes,
+        uint128[] memory prices,
+        uint16[] memory tokenIdsForPrice
+    ) internal {
+        previousSolutionBatchId = batchIndex;
+        for (uint i = 0; i < owners.length; i++) {
+            previousSolutionTrades.push(TradeData({
+                owner: owners[i],
+                orderId: orderIds[i],
+                volume: volumes[i]
+            }));
+        }
+        for (uint i = 0; i < prices.length; i++) {
+            previousSolutionTokenIdForPrice.push(tokenIdsForPrice[i]);
```

Comment:
Can you not assign the array directly?

`previousSolutionTokenIdForPrice = tokenIdsForPrice`